PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h

```diff
@@ -885,38 +1237,51 @@ TH1D* GetHistogramWithWeights(const char* filePath, const char* variable)
     exit(1);
   }
 
-  // c) Check if the external ROOT file exists at specified path:
-  /*
- if(gSystem->AccessPathName(filePath,kFileExists))
- {
-  //Red(Form("if(gSystem->AccessPathName(filePath,kFileExists)), filePath = %s",filePath)); // use eventually
-  cout<<__LINE__<<endl;
-  exit(1);
- }
- */
-
-  // d) Access the external ROOT file and fetch the desired histogram with weights:
-  //TFile *weightsFile = TFile::Open(filePath,"READ");
-  TGrid* alien = TGrid::Connect("alien", gSystem->Getenv("USER"), "", "");
-  if (!alien) {
-    cout << __LINE__ << endl;
-    exit(1);
+  // c) Determine from filePath if the file in on a local machine, or in AliEn:
+  //    Algorithm: If filePath begins with "/alice/cern.ch/" then it's in AliEn.
+  //               Therefore, files on AliEn must be specified with abs path, for local files both abs and relative paths are fine.
+  Bool_t bFileIsInAliEn = kFALSE;
+  if (TString(filePath).BeginsWith("/alice/cern.ch/")) {
+    bFileIsInAliEn = kTRUE;
   }
-  TFile* weightsFile = TFile::Open(Form("alien://%s", filePath), "READ");
 
+  // d) Handle the AliEn case:
+  TFile* weightsFile = NULL;
+  if (bFileIsInAliEn) {
+    TGrid* alien = TGrid::Connect("alien", gSystem->Getenv("USER"), "", "");
+    if (!alien) {
+      cout << __LINE__ << endl;
+      exit(1);
+    }
+    weightsFile = TFile::Open(Form("alien://%s", filePath), "READ");
+  } else {
+    // e) Handle the local case:
+
+    // Check if the external ROOT file exists at specified path:
+    if (gSystem->AccessPathName(filePath, kFileExists)) {
+      Red(Form("if(gSystem->AccessPathName(filePath,kFileExists)), filePath = %s", filePath));
+      cout << __LINE__ << endl;
+      exit(1);
+    }
+    weightsFile = TFile::Open(filePath, "READ");
+  } // if(bFileIsInAliEn)
+
+  // f) Access the external ROOT file and fetch the desired histogram with weights:
```

Comment:
I don't think it is intended to open general `alien` access from the grid nodes
At least for corrections files you should consider using CCDB